Several API updates for `resource` & friends from the RFC

alxhub opened the pull request.

alxhub committed:
refactor(core): convert `ResourceStatus` to a string type

An outcome of the Resource RFC was that we should use string constants for
communicating the resource status instead of an enum. This commit converts
`ResourceStatus` accordingly.

alxhub committed:
refactor(core): rename resource's `request` to `params`

As decided in the resource RFC, this commit renames the `request` option of
a resource to `params`, including the subsequent argument passed to the
loader. It also corrects the type in the process to properly allow narrowing
of the `undefined` value.

Fixes #58871

alxhub committed:
refactor(core): rename `loader` parameter to `stream` for `rxResource`

As decided in the Resource RFC, this commit renames `loader` in `rxResource`
to `stream`. Some logic is left in to support gradual rollout of this change
in g3.

mmalerba reviewed (approved):
reviewed-for: public-api

thePunderWoman reviewed (approved):
LGTM

reviewed-for: fw-general, angular-dev, public-api

alxhub committed:
refactor(common): delete unused code for `HttpResource`

Remove the code related to exposing the response of an HTTP request as a
`Resource` itself, as the final API will not go in this direction.

alxhub commented:
Caretaker: this requires a g3 patch and g3 updates from cl/750662376.
